pandas/_typing.py

```diff
@@ -22,3 +22,5 @@
                              Timedelta)
 Dtype = Union[str, np.dtype, ExtensionDtype]
 FilePathOrBuffer = Union[str, Path, IO[AnyStr]]
```

Comment:
@WillAyd we should put a line comment on when to use these / what these are if not completely obvious (and then maybe even so).

future PR of course

Reply:
Yea I'm going to take another look at this module soon to see if there's a better way of doing things. Also wondering if we should use `if TYPE_CHECKING` imports in that module instead of the ABCs. Will check it out and post separately

Reply:
> Yea I'm going to take another look at this module soon to see if there's a better way of doing things. Also wondering if we should use `if TYPE_CHECKING` imports in that module instead of the ABCs. Will check it out and post separately

-1 on this. I think the way you have it setup is very nice actually, unless I am missing something big.

Reply:
My concern is that the ABC* classes aren't getting type checked correctly. Adding `reveal_type(klass)` to this PR where the TypeVar is used yields the following:

```sh
$ mypy pandas/core/groupby/generic.py
pandas/core/groupby/generic.py:83: error: Revealed type is 'Type[Any]'
```

That coupled with @topper-123 's comment around code completion makes me think the ABCs might all be `Type[Any]` since they are generated by `create_pandas_abc_type`, a function without any annotations itself (so implicitly `Type[Any]`)

Going to investigate and open a separate issue accordingly

Reply:
ok fair. I just don't want to introduce overhead for contributors, which typing especially with 'non-standard' things (e.g. ``cast``) can do; we really want to minimize what people have to do (nothwithstanding maybe we have magic that makes things easy).